source/extensions/omni.isaac.lab/omni/isaac/lab/assets/rigid_object_collection/rigid_object_collection_data.py

```diff
@@ -0,0 +1,251 @@
+
+        Args:
+            root_physx_view: The root rigid body view.
+            num_objects: The number of objects in the collection.
+            device: The device used for processing.
+        """
+        # Set the parameters
+        self.device = device
+        self.num_objects = num_objects
+        # Set the root rigid body view
+        # note: this is stored as a weak reference to avoid circular references between the asset class
+        #  and the data container. This is important to avoid memory leaks.
+        self._root_physx_view: physx.RigidBodyView = weakref.proxy(root_physx_view)
+        self.num_instances = self._root_physx_view.count // self.num_objects
+
+        # Set initial time stamp
+        self._sim_timestamp = 0.0
+
+        # Obtain global physics sim view
+        physics_sim_view = physx.create_simulation_view("torch")
+        physics_sim_view.set_subspace_roots("/")
+        gravity = physics_sim_view.get_gravity()
+        # Convert to direction vector
+        gravity_dir = torch.tensor((gravity[0], gravity[1], gravity[2]), device=self.device)
+        gravity_dir = math_utils.normalize(gravity_dir.unsqueeze(0)).squeeze(0)
+
+        # Initialize constants
+        self.GRAVITY_VEC_W = gravity_dir.repeat(self.num_instances, self.num_objects, 1)
+        self.FORWARD_VEC_B = torch.tensor((1.0, 0.0, 0.0), device=self.device).repeat(
+            self.num_instances, self.num_objects, 1
+        )
+
+        # Initialize the lazy buffers.
+        self._object_state_w = TimestampedBuffer()
+        self._object_acc_w = TimestampedBuffer()
+
+    def update(self, dt: float):
+        """Updates the data for the rigid object.
+
+        Args:
+            dt: The time step for the update. This must be a positive value.
+        """
+        # update the simulation timestamp
+        self._sim_timestamp += dt
+
+    ##
+    # Names.
+    ##
+
+    object_names: list[str] = None
+    """Object names in the order parsed by the simulation view."""
+
+    ##
+    # Defaults.
+    ##
+
+    default_object_state: torch.Tensor = None
+    """Default object state ``[pos, quat, lin_vel, ang_vel]`` in local environment frame. Shape is (num_instances, num_objects, 13).
+
+    The position and quaternion are of each object's rigid body's actor frame. Meanwhile, the linear and angular velocities are
+    of the center of mass frame.
+    """
+
+    default_mass: torch.Tensor = None
+    """Default object mass read from the simulation. Shape is (num_instances, num_objects, 1)."""
+
+    default_inertia: torch.Tensor = None
+    """Default object inertia tensor read from the simulation. Shape is (num_instances, num_objects, 9).
+
+    The inertia is the inertia tensor relative to the center of mass frame. The values are stored in
+    the order :math:`[I_{xx}, I_{xy}, I_{xz}, I_{yx}, I_{yy}, I_{yz}, I_{zx}, I_{zy}, I_{zz}]`.
+    """
+
+    ##
+    # Properties.
+    ##
+
+    @property
+    def object_state_w(self):
```

Comment:
Do we want to use the same convention as RigidObjects for these properties i.e. `body_`? Or do we want to change RigidObjects to reflect this convention? What are your thoughts?

Reply:
This is a question I have been thinking a lot about when designing this.

The core difference between rigid objects, and object collections is that object collections can have rigid bodies that contain the same prim structure (for example when two rigid bodies come from the same USD). That is why something like `body_names` would not work in general as there would be duplicates in that case. For that reason, I have introduced the concept of `object_names` which refer to the keys of the dict you create the config with (see [code](https://github.com/isaac-sim/IsaacLab/blob/feature/rigid_object_collection/source/extensions/omni.isaac.lab/omni/isaac/lab/assets/rigid_object_collection/rigid_object_collection_cfg.py#L24-L27)). This allows you to uniquely identify each rigid object in the collection.

I thought to be consistent with that, I would rename all the occurrences of `body` with `object` to mark that difference. 

But I am open for debate :)

Reply:
Ah OK I see what you mean. I think that is reasonable.

Reply:
Let's see what @Mayankm96 / @kellyguo11 say

Reply:
I think the distinction makes sense, maybe we just need to make the use cases of these classes clear in the docs so users understand the differences between the classes.